src/app/canvas/Canvas.tsx

```diff
@@ -0,0 +1,128 @@
+'use client';
+
+import { useState, useCallback } from 'react';
+import update from 'immutability-helper';
+import { Stack } from '@mui/material';
+import { TaskCard } from '@/components/cards/TaskCard';
+import { TaskColumn } from '@/components/cards/TaskColumn';
+import { Droppable } from '@/hoc/Droppable';
+
+interface Task {
+  assignedTo: string;
+  id: number;
+}
+
+interface Lists {
+  [key: number]: Task[];
+}
+
+const mockTaskTodoData: Task[] = [
+  {
+    assignedTo: 'ragnar',
+    id: 1,
+  },
+  {
+    assignedTo: 'john',
+    id: 2,
+  },
+  {
+    assignedTo: 'yuki',
+    id: 3,
+  },
+];
+
+const mockTaskInProgressData: Task[] = [
+  {
+    assignedTo: 'doe',
+    id: 4,
+  },
+  {
+    assignedTo: 'floki',
+    id: 5,
+  },
+  {
+    assignedTo: 'professor',
+    id: 6,
+  },
+];
+
+const mockCompletedTask: Task[] = [
+  {
+    assignedTo: 'rock',
+    id: 7,
+  },
+];
+
+export const Canvas = () => {
+  const [lists, setLists] = useState<Lists>({
+    1: mockTaskTodoData,
+    2: mockTaskInProgressData,
+    3: mockCompletedTask,
+  });
+
+  const moveCard = useCallback(
+    (dragIndex: number, hoverIndex: number, sourceId: number, targetId: number) => {
+      const sourceList = lists[sourceId];
+      const targetList = lists[targetId];
+
+      if (sourceId === targetId) {
+        const updatedList = update(sourceList, {
```

Comment:
what is this doing? is it rearrange within the column?

Reply:
`sourceId` is the ID of the list from where the card is being moved (origin) and `targetId` is the ID of the list where the task is being moved (destination). It is checking if the task is getting moved in the same list or the another list.